packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -768,4 +953,81 @@ export class IntervalCollection<TInterval extends ISerializableInterval> {
         // eslint-disable-next-line @typescript-eslint/no-unsafe-return
         return this.view.serializeInternal();
     }
+
+    public [Symbol.iterator](): IntervalCollectionIterator<TInterval> {
+        const iterator = new IntervalCollectionIterator<TInterval>(this);
+        return iterator;
+    }
+
+    public CreateForwardIteratorWithStartPosition(startPosition: number): IntervalCollectionIterator<TInterval> {
```

Comment:
rather than putting all these methods on the object itself these could instead be static create methods on the interator class itself. i prefer the keep the interface on this smaller, to reduce compat our compat burden if/when update the interface to be more in line with our current dds patterns (no view model split)

Reply:
I see what you mean. A static method can't refer to the type parameter TInterval, though, so would we need separate static methods for each interval type (Interval, SequenceInterval, etc.)?

Reply:
ahhh you are right. i didn't notice the type param.

Reply:
I'd be fine, though, with reducing the methods to CreateForwardIterator and CreateBackwardIterator, each taking start?:number and end?:number.

Reply:
i'll let you make the call. i slightly prefer having start and end, but no strong opinion